allennlp/data/dataset.py

```diff
@@ -77,7 +77,8 @@ def get_padding_lengths(self) -> Dict[str, Dict[str, int]]:
 
     def as_arrays(self,
                   padding_lengths: Dict[str, Dict[str, int]] = None,
-                  verbose: bool = True) -> Dict[str, List[numpy.array]]:
+                  verbose: bool = True) -> Dict[str, List[Union[numpy.array,
```

Comment:
Also @joelgrus, does mypy now pass on the whole codebase?  If so, can we add it to our travis set up?